site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,574 @@
+| [Semantic UI (uses checkbox)](https://semantic-ui.com/modules/checkbox.html)                                                                                                                                           | checkbox        |                             |                         |                ✅ |
+| [Spectrum (uses checkbox)](https://spectrum.adobe.com/page/switch/)                                                                                                                                                    | checkbox        |                             |                         |                ✅ |
+| [Tailwind Elements (uses checkbox)](https://tailwind-elements.com/docs/standard/forms/switch/)                                                                                                                         | checkbox        |                             |                         |                ✅ |
+| [WAI-ARIA (examples include button and checkbox variant)](https://www.w3.org/WAI/ARIA/apg/patterns/switch/)                                                                                                            | button/checkbox |                             |                         |                ✅ |
+
+_\* No Data found for explicit form usage encouragement, allthough implementations that rely on a checkbox indirectly allow that behavior._
+
+#### Pattern differentiation
+
+The switch is not a checkbox or radiobutton alternative.
+
+A switch differs from a checkbox because:
+
+- It has no indeterminate state
+- It requires an immediate action, no confirmation is needed for the state to take effect
+
+A switch differs from a radio button because:
+
+- It always has an initial state, either on or off. A radio button group can be in an initial "nothing selected", invalid state
+- Switches cannot be grouped together like radio buttons with the same `name` attribute. Two switches cannot have the same `name` attribute.
+- It requires an immediate action, no confirmation is needed for the state to take effect
+
+### Features
+
+{/* _A list of the key features unique to this component._ */}
+
+#### Interaction
+
+- `click`/`touch`: the switch can be toggled by clicking it
+- `drag&drop`: the switch can be toggled by dragging the thumb
+
+#### Form association
+
+<div class="question">
+  How should form validation work? Is there a way to describe that the valid state is "off" instead
+  of the more intuitive "on"?
+</div>
+
+A `<switch>` can be part of a form and has a mechanism (`type="submit"`) to immediately submit the form it's part of. The value of a switch is reported to the form target as either "on" or "off".
```

Comment:
It's an attempt to provide this idea of "switches trigger an immediate action" to non-javascript use cases. A simple example could be toggling dark mode server side on a html only page

```html
<form action="/index.html?mode=dark">
  <label for="dark-mode">Turn dark mode on</label>
  <switch id="dark-mode" type="submit"></switch>
</form>
```

I don't know if this is a good idea, it's something that I'd open a ticket for so it can be discussed.

Reply:
Auto-submission is an anti-pattern IMO. I don't think we want to encourage it. If you think it is necessary please do raise an issue, but I would object to this for the reasons Scott has already mentioned.

Reply:
Thanks for the reply. But yeh, agree with Keith. Tbh, in a no js scenario I’d rather someone used a noscript that contained a form consisting of a checkbox and a submit button.   Don’t blur the lines with switch acting like a checkbox and a submit where type=submit could be used with or without JS

Reply:
I removed the part about the switch being able to submit forms.